spec.bs

```diff
@@ -130,110 +131,91 @@ are:
     is not in the range [0, 2<sup>128</sup>−1].
 1. Throw {{RangeError}} if |contribution|["{{PAHistogramContribution/value}}"]
     is negative.
-1. [=list/Append=] |contribution| to the [=contributions cache=].
-
-Issue: Check that value can actually be zero in the spec pipeline.
-
+1. Let |batchingScopeSteps| be the {{PrivateAggregation}} object's
+    [=PrivateAggregation/get batching scope steps=].
+1. If |batchingScopeSteps| is null, throw a new {{DOMException}} with name
+    "`NotAllowedError`".
 
-Exposing to the Shared Storage API {#shared-storage}
-====================================================
-
-<xmp class="idl">
-partial interface SharedStorageWorkletGlobalScope {
-  readonly attribute PrivateAggregation privateAggregation;
-};
-</xmp>
+    Note: This indicates the API is not yet available, for example, because the
+        the initial execution of the script after loading is not complete.
 
-Immediately after an operation completes, [=process the Shared Storage
-contributions cache=] given the worklet's [=contributions cache=] and the
-worklet's global scope.
+    Issue: Consider improving developer ergonomics here (e.g. a way to detect
+        this case).
 
-Issue: How to handle batching different operation invocations properly. This
-    doesn't work for simultaneous operations. Also disallow usage outside an
-    operation.
+1. Let |entry| be a new [=contribution cache entry=] with the items:
+    : [=contribution cache entry/contribution=]
+    :: |contribution|
+    : [=contribution cache entry/batching scope=]
+    :: The result of running |batchingScopeSteps|
 
-To <dfn>process the Shared Storage contributions cache</dfn> given a [=list=]
-|contributionsCache| and a {{SharedStorageWorkletGlobalScope}} |scope|, perform
-the [=PrivateAggregation/report creation and scheduling steps=] with |scope|'s
-[=relevant settings object=]'s [=environment settings object/origin=],
-"<code>[=context type/shared-storage=]</code>" and |contributionsCache|.
+1. [=Append an entry to the contribution cache=] given |entry|.
```

Comment:
Maybe we can do something like
1. [=Append|Append an entry to the contribution cache=] |entry| to the [=contribution cache=].

Reply:
Done